sw/host/opentitanlib/src/otp/otp_mmap.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::otp::otp::Wrapped;
+use crate::otp::vmem_serialize::VmemSerialize;
+use crate::otp::otp_img::OtpImg;
+
+use anyhow::{anyhow, Result};
+use rand::Fill;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+
+#[derive(Deserialize, Debug)]
+struct OtpMapConfig {
+    width: Wrapped<isize>,
+    depth: Wrapped<isize>,
```

Comment:
Depending on whether or not you need to keep the original string inside the "wrapped" value, there are some alternatives with serde that might be easier to implement:

1. [`#[serde(serialize_with="...")]`](https://serde.rs/field-attrs.html#serialize_with) and friends allow you to write a more simplified serialize/deserialize interface for individual fields.  The advantage is that the native field type in the struct can be `isize` or `u64` (or whatever) and the custom serialize/deserialize function can deal with converting to whatever is required (ie: quoted string in hex representation).

   You can see a simple example of this technique [here](https://github.com/cfrantz/z2doc/blob/master/disassembler/src/description/nesfile.rs#L33) with the corresponding implementation in [`range.rs`](https://github.com/cfrantz/z2doc/blob/master/disassembler/src/description/range.rs).

2. For container types (ie: structs), you can use [`#[serde(from="...")]`](https://serde.rs/field-attrs.html#serialize_with) and `into` to ask serde to convert the type to some other type (via From/Into traits) before serialization.  You can then write simpler `From` implementations between the in-rust representation and the preferred serialization representation and the let serde "do the right thing" with the serialization-representation struct.